firebase/rules/tests/handshakes.spec.ts

```diff
@@ -0,0 +1,114 @@
+import 'jest';
+import * as matchers from '../matchers';
+import * as targaryen from 'targaryen';
+
+import { SYSTEM_USER } from '@machinelabs/models';
+
+const anonymousUser = targaryen.util.users.anonymous;
+const rules = require('../database.rules.json');
+
+describe('/handshakes', () => {
+  const currentUser = { uid: '1', email: 'foo@bar.com', provider: 'github' };
+  const systemUser = { uid: SYSTEM_USER };
+
+  const testHandshakeRequest = {
+    id: '1',
+    token: null
+  };
+
+  const testHandshakeCommit = {
+    id: '1',
+    user_id: '1'
+  };
+
+  beforeEach(() => {
+    jest.addMatchers(matchers);
+
+    targaryen.util.setFirebaseData({
+      handshakes: {
+        '1': {
+          request: {
+            id: '1',
+            token: null
+          }
+        },
+        '2': {
+          request: {
+            id: '2',
+            token: 'some-token'
+          },
+          commit: {
+            id: '2',
+            user_id: '1'
+          }
+        }
+      }
+    });
+
+    targaryen.util.setFirebaseRules(rules);
+  });
+
+  it('Only System-user can read into all handshake requests', () => {
+    expect(anonymousUser).cannotRead('/handshakes/2/request');
+    expect(currentUser).cannotRead('/handshakes/2/request');
+    expect(systemUser).canRead('/handshakes/2/request');
+  });
```

Comment:
According to database.rules

```
path /handshakes/{id}/request is HandshakeRequest {
  read() { isCurrentUser(id) }
  write() { isCurrentUser(id) }
}
```

Only the current authenticated users can read (and write)

Reply:
As you can see, the `id` param in the test url is 2, and since the `isCurrentUser` test checks this `id` param against the `uid` of the user, currentUser with `id  = 1` shouldn't be allowed to read into this

Reply:
Yeap, makes sense. It's just not super obvious when looking at the test like that. But I dig it.

Reply:
Yea... I agree but can't really change much about that sadly :-)